### PR TITLE
chore: switch to use gcr images

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
chore: switch to use gcr images

- image build actually works now:
```
I0504 02:57:57.973970       1 nfs.go:60] Driver: nfs.csi.k8s.io version: 3.0.0
I0504 02:57:57.974003       1 nfs.go:124] Enabling volume access mode: SINGLE_NODE_WRITER
I0504 02:57:57.974007       1 nfs.go:124] Enabling volume access mode: SINGLE_NODE_READER_ONLY
I0504 02:57:57.974010       1 nfs.go:124] Enabling volume access mode: MULTI_NODE_READER_ONLY
I0504 02:57:57.974013       1 nfs.go:124] Enabling volume access mode: MULTI_NODE_SINGLE_WRITER
I0504 02:57:57.974016       1 nfs.go:124] Enabling volume access mode: MULTI_NODE_MULTI_WRITER
I0504 02:57:57.974019       1 nfs.go:135] Enabling controller service capability: CREATE_DELETE_VOLUME
I0504 02:57:57.974022       1 nfs.go:145] Enabling node service capability: GET_VOLUME_STATS
I0504 02:57:57.974025       1 nfs.go:145] Enabling node service capability: UNKNOWN
I0504 02:57:57.974493       1 nfs.go:107]
DRIVER INFORMATION:
-------------------
Build Date: "2021-05-03T13:47:06Z"
Compiler: gc
Driver Name: nfs.csi.k8s.io
Driver Version: v3.0.0
Git Commit: ""
Go Version: go1.16
Platform: linux/amd64

Streaming logs below:
I0504 02:57:57.974564       1 mount_linux.go:192] Detected OS without systemd
I0504 02:57:57.975406       1 server.go:117] Listening for connections on address: &net.UnixAddr{Name:"//csi/csi.sock", Net:"unix"}
I0504 02:57:58.393771       1 utils.go:86] GRPC call: /csi.v1.Identity/GetPluginInfo
I0504 02:57:58.393791       1 utils.go:87] GRPC request: {}
I0504 02:57:58.394880       1 utils.go:93] GRPC response: {"name":"nfs.csi.k8s.io","vendor_version":"3.0.0"}
I0504 02:57:58.644604       1 utils.go:86] GRPC call: /csi.v1.Identity/GetPluginInfo
I0504 02:57:58.644619       1 utils.go:87] GRPC request: {}
I0504 02:57:58.644658       1 utils.go:93] GRPC response: {"name":"nfs.csi.k8s.io","vendor_version":"3.0.0"}
I0504 02:57:59.262428       1 utils.go:86] GRPC call: /csi.v1.Node/NodeGetInfo
I0504 02:57:59.262442       1 utils.go:87] GRPC request: {}
I0504 02:57:59.262480       1 utils.go:93] GRPC response: {"node_id":"aks-nodepool1-18870927-vmss000003"}
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

cc @msau42 

**Does this PR introduce a user-facing change?**:

```release-note
chore: switch to use gcr images
```

### CODE REVIEW

File: charts/latest/csi-driver-nfs/values.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 image:
     nfs:
-        repository: mcr.microsoft.com/k8s/csi/nfs-csi
```
Comment (msau42) [inline charts/latest/csi-driver-nfs/values.yaml:3]: Images in k8s-staging are only kept for 30 days before being deleted. It's better to use a tagged version from k8s.gcr.io/sig-storage. If we plan to make a release soon, then I think it's fine to use a tag that doesn't exist yet.

Comment (andyzhangx) [reply charts/latest/csi-driver-nfs/values.yaml:3]: `gcr.io/k8s-staging-sig-storage/nfsplugin:canary` image is built every day, I think that's ok for master branch, user could always use this image built from master branch

**Review by msau42 — COMMENTED** [review]
What is the csi-driver-nfs-v3.0.0.tgz file?

File: charts/latest/csi-driver-nfs/values.yaml (hunk 1 above)
Comment (msau42) [reply charts/latest/csi-driver-nfs/values.yaml:3]: Can we call it "unstable" or "dev" to make it more obvious that these images are not officially released versions and they should not rely on this for production.

The canary images are rebuilt when a PR merges. There is a job that refreshes the image every week as well but if that job fails for some reason and we don't notice/fix it, the default image deletion policy will kick in.

**Pull request closed without merging**